src/osyris/spatial/coordinate_transforms.py

```diff
@@ -0,0 +1,65 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/osyris-project/osyris)
+
+import numpy as np
+from . import utils
+
+
+def change_origin(dataset, new_origin):
+    """
+    Translate all positionnal coordinates to new origin
+    """
+    for g in dataset.groups.keys():
+        for element in dataset[g]:
+            unit = dataset[g][element].unit
+            if unit.is_compatible_with("meter") and hasattr(dataset[g][element],
+                                                            "nvec"):
+                dataset[g][element] -= new_origin
+    dataset.origin = new_origin
+
+
+def rotation_matrix(vec, angle):
```

Comment:
Does this need to be available to the user? Or is it only unsed internally?

Reply:
For the moment it's only used internally. Maybe I can make it private and move it to utils until the cartopy projections PR. The user might want to use it to rotate their cartopy projection by a specific angle around a specific vector, which could be more intuitive in certain cases than using the yaw/pitch/roll rotation matrix.